x-pack/elastic-agent/pkg/core/monitoring/server/process.go

```diff
@@ -0,0 +1,163 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/elastic/beats/v7/metricbeat/mb/parse"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/program"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/artifact"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/monitoring/beats"
+)
+
+const (
+	processIDKey      = "processID"
+	monitoringSuffix  = "-monitoring"
+	separator         = "-"
+	timeout           = 10 * time.Second
+	errTypeUnexpected = "UNEXPECTED"
+
+	httpPlusPrefix = "http+"
+)
+
+var (
+	// ErrProgramNotSupported returned when requesting metrics for not supported program.
+	ErrProgramNotSupported = errors.New("specified program is not supported")
+)
+
+func processHandler() func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+		vars := mux.Vars(r)
+		id, found := vars[processIDKey]
+
+		if !found {
+			writeResponse(
+				w,
+				unexpectedErrorWithReason("productID not found"),
+			)
```

Comment:
We should also make use of status codes instead of messages. When we update metricbeat to collect these metrics it should be able to handle a `404`. Using the API requires multiple HTTP requests and the overall collection process contains race conditions. If an ID does not exist anymore, metricbeat will must ignore it or produce a special event indicating that the process was down.

There are other potential status codes like 400, 408, 409, 410, ...
In case of a process being removed thanks to reconfiguration it will be 404. In case of a restart we might want to retry and produce a 'timeout' or 'resource gone' status code to indicate some temporary condition.